dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1290,18 +1290,29 @@ public Workflow manualRegister(@ApiParam(hidden = true) @Parameter(hidden = true
                 if (!wasFrozen && nowFrozen) {
                     Optional<String> toolsJSONTable;
                     LanguageHandlerInterface lInterface = LanguageHandlerFactory.getInterface(w.getFileType());
-                    // Store tool table json
-                    if (existingTag.getToolTableJson() == null) {
+
+                    String existingToolTableJson = existingTag.getToolTableJson();
+                    if (existingToolTableJson != null && (existingToolTableJson.contains("specifier") || "[]".equals(existingToolTableJson))) {
+                        toolsJSONTable = Optional.of(existingToolTableJson);
+                    } else {
+                        // Store tool table json
                         toolsJSONTable = lInterface.getContent(w.getWorkflowPath(), getMainDescriptorFile(existingTag).getContent(), extractDescriptorAndSecondaryFiles(existingTag), LanguageHandlerInterface.Type.TOOLS, toolDAO);
                         existingTag.setToolTableJson(toolsJSONTable.get());
-                    } else {
-                        toolsJSONTable = Optional.of(existingTag.getToolTableJson());
                     }
 
                     if (toolsJSONTable.isPresent()) {
+                        try {
+                            // Check that a snapshot can occur (all images are referenced by tag or digest)
+                            lInterface.checkSnapshotImages(existingTag.getName(), toolsJSONTable.get());
+                        } catch (CustomWebApplicationException ex) {
+                            existingTag.setFrozen(false);
```

Comment:
It doesn't hurt, but I don't think you need this either -- the throw is going to cause the transaction to be rolled back. If you don't need it, then you can get rid of the try catch, and just let 1306 throw